tests/unit/utils/test_objectives.py

```diff
@@ -79,6 +81,104 @@ def test_logarithmic_goldstein_price_no_function_values_are_less_than_global_min
     )
 
 
+@pytest.mark.parametrize(
+    "test_x, expected",
+    [
+        (
+            tf.constant([[0.0, 0.0]]),
+            tf.convert_to_tensor([[1.0 - tf.math.exp(-1.0), 1.0 - tf.math.exp(-1.0)]]),
+        ),
+        (
+            tf.constant([[0.5, 1.0]]),
+            tf.convert_to_tensor(
+                [
+                    [
+                        1.0
+                        - tf.math.exp(
+                            -((0.5 - 1 / tf.sqrt(2.0)) ** 2) - (1.0 - 1 / tf.sqrt(2.0)) ** 2
+                        ),
+                        1.0
+                        - tf.math.exp(
+                            -((0.5 + 1 / tf.sqrt(2.0)) ** 2) - (1.0 + 1 / tf.sqrt(2.0)) ** 2
+                        ),
+                    ]
+                ]
+            ),
+        ),
+    ],
+)
+def test_vlmop2_has_expected_output(test_x: TensorType, expected: TensorType):
+    npt.assert_allclose(vlmop2(test_x), expected, rtol=1e-5)
+
+
+@pytest.mark.parametrize(
+    "test_x, input_dim, num_obj, expected",
+    [
+        (tf.constant([[0.0, 0.2, 0.4]]), 3, 2, tf.constant([[0.0, 5.5]])),
+        (tf.constant([[0.8, 0.6, 0.4, 0.2]]), 4, 2, tf.constant([[4.8, 1.2]])),
+        (tf.constant([[0.1, 0.2, 0.3, 0.4]]), 4, 3, tf.constant([[0.06, 0.24, 2.7]])),
+    ],
+)
+def test_dtlz1_has_expected_output(
+    test_x: TensorType, input_dim: int, num_obj: int, expected: TensorType
+):
+    f = DTLZ1(input_dim, num_obj).prepare_benchmark()
+    npt.assert_allclose(f(test_x), expected, rtol=1e-5)
+
+
+@pytest.mark.parametrize(
+    "test_x, input_dim, num_obj, expected",
+    [
+        (tf.constant([[0.0, 0.2, 0.4]]), 3, 2, tf.constant([[1.1, 0.0]])),
+        (tf.constant([[0.8, 0.6, 0.4, 0.2]]), 4, 2, tf.constant([[0.3430008637, 1.055672733]])),
+        (
+            tf.constant([[0.1, 0.2, 0.3, 0.4]]),
+            4,
+            3,
+            tf.constant([[0.9863148, 0.3204731, 0.16425618]]),
+        ),
+    ],
+)
+def test_dtlz2_has_expected_output(
+    test_x: TensorType, input_dim: int, num_obj: int, expected: TensorType
+):
+    f = DTLZ2(input_dim, num_obj).prepare_benchmark()
+    npt.assert_allclose(f(test_x), expected, rtol=1e-4)
+
+
+@pytest.mark.parametrize(
+    "obj_inst, input_dim, num_obj, gen_pf_num",
+    [
+        (DTLZ1(3, 2), 3, 2, 1000),
+        (DTLZ1(5, 3), 5, 3, 1000),
+        (DTLZ2(3, 2), 3, 2, 1000),
+        (DTLZ2(12, 6), 12, 6, 1000),
+    ],
+)
+def test_gen_pareto_front_is_equal_to_math_defined(
```

Comment:
Also should vlomp also have some tests in here?

Reply:
It seems it is rather difficult for vlmop have test for pareto front. As this test is mainly used for testing if the algorithm generated pareto front is equivalent to math defined (e.g., a simplex for DTLZ1 and hypersphere for DTLZ2, which means the PF in objective space have some math defined relationship that can be checked). 
While for vlmop2, according to the citation, the pareto front for vlmop is just defined according to input distributions (i.e., "True pareto lies on -1/sqrt(2)-1/sqrt(2) and x1=x2"), which also conveys the PF do not have any analytical expression, hence might be hard to test.